src/connection.js

```diff
@@ -31,15 +31,31 @@ module.exports = function connection (swarm) {
         //   3. add this conn to the pool
         if (swarm.identify) {
           // overload peerInfo to use Identify instead
+          console.log('-> swarm.identify')
           conn.getPeerInfo = (cb) => {
+            console.log('-> getPeerInfo gets called')
             const conn = muxedConn.newStream()
             const ms = new multistream.Dialer()
 
+            console.log('waterfall')
             waterfall([
-              (cb) => ms.handle(conn, cb),
-              (cb) => ms.select(identify.multicodec, cb),
-              (conn, cb) => identify.dialer(conn, cb),
+              (cb) => {
+                console.log('ms-handle')
+                console.log(conn)
+                console.log('\n')
+                setTimeout(() => ms.handle(conn, cb), 2000)
```

Comment:
nvm, I just didn't increase the timeout on spdy for the checking. Still remains an issue with multiplex.